smalltalksrc/Slang-Tests/SlangBasicTranslationTest.class.st

```diff
@@ -1618,6 +1675,187 @@ SlangBasicTranslationTest >> testSendCppIfIfTrueIfFalse [
+					         (TStmtListNode new setStatements: { 
+							          (TAssignmentNode new
+								           setVariable: (TVariableNode new setName: 'a')
+								           expression: (TConstantNode value: true)).
+							          (TAssignmentNode new
+								           setVariable: (TVariableNode new setName: 'c')
+								           expression: (TConstantNode value: 0)) }) }) }.
+	translation := self translate: send.
+
+	self assert: translation trimBoth equals: '{
+	
+#  if FEATURE
+	{
+		a = 0;
+		c = 1;
+	}
+#  else /* FEATURE */
+	{
+		a = 1;
+		c = 0;
+	}
+#  endif /* FEATURE */
+}'
+]
+
+{ #category : #'tests-builtins' }
+SlangBasicTranslationTest >> testSendCppIfIfTrueIndentationInBlock [
+
+	"self cppIf: #FEATURE ifTrue: [ a := false. c := 1 ]"
+
+	| translation send |
+	send := TStmtListNode new setStatements:
+		        { (TStmtListNode new setStatements: { (TSendNode new
+				          setSelector: #cppIf:ifTrue:
+				          receiver: (TVariableNode new setName: 'self')
+				          arguments: { 
+						          (TConstantNode value: #FEATURE).
+						          (TStmtListNode new setStatements: { 
+								           (TAssignmentNode new
+									            setVariable: (TVariableNode new setName: 'a')
+									            expression: (TConstantNode value: false)).
+								           (TAssignmentNode new
+									            setVariable: (TVariableNode new setName: 'c')
+									            expression: (TConstantNode value: 1)) }) }) }) }.
+	translation := self translate: send.
+
+	self assert: translation trimBoth equals: '{
+	{
+		
+#    if FEATURE
+		{
+			a = 0;
+			c = 1;
+		}
+#    endif /* FEATURE */
+	}
+}'
+]
+
+{ #category : #'tests-builtins' }
+SlangBasicTranslationTest >> testSendCppIfIfTrueWithSendIfTrueAsCondition [
+
+	"self cppIf: (true ifTrue: [ false ] ifFalse: [ true ]) ifTrue: [ a := false. c := 1 ]"
+
+	| translation send |
+	generator generateDeadCode: false.
+	send := TSendNode new
+		        setSelector: #cppIf:ifTrue:
+		        receiver: (TVariableNode new setName: 'self')
+		        arguments: { 
+				        (TSendNode new
+					         setSelector: #ifTrue:ifFalse:
+					         receiver: (TConstantNode value: true)
+					         arguments: { 
+							         (TStmtListNode new setStatements:
+								          { (TConstantNode value: false) }).
+							         (TStmtListNode new setStatements:
+								          { (TConstantNode value: true) }) }).
+				        (TStmtListNode new setStatements: { 
+						         (TAssignmentNode new
+							          setVariable: (TVariableNode new setName: 'a')
+							          expression: (TConstantNode value: false)).
+						         (TAssignmentNode new
+							          setVariable: (TVariableNode new setName: 'c')
+							          expression: (TConstantNode value: 1)) }) }.
+	translation := self translate: send.
+
+	self assert: translation trimBoth equals: '{
+	a = 0;
+	c = 1;
+}'
```

Comment:
This is weird. If you look at the comment, the part "true ifTrue: [ false ] ifFalse: [ true ]" is evaluated as false, but Slang checks if the receiver (which is a true constant) of the ifTrue:ifFalse: message is a boolean constant. Why the receiver ? It shouldn't print anything in this case, right ?

Reply:
Let's give a quick look tomorrow. I'd say you've found a bug :)